### ISSUE TITLE
Workflow does not support iCloud vault path

### ISSUE THREAD

**Issue by KingOfSpades** [macedotavares/obsidian-utilities#10]
Hey,

Not sure this is something that can be fixed. I think it's on the Apple side of things or it is my own doing. Lemme explain :)

I'm using iCloud to sync/store my Obsidian vault. To make things easy for me I created a Synlink to the vault in my personal folder. This looks like:

`/Users/name/ZETTL/ZETTLKST` -> `/Users/name/Library/Mobile Documents/com~apple~CloudDocs/ZETTL/ZETTLKST`

Obsidian also sees the vault @ /Users/name/ZETTL/ . Now when I use Obsidian utilities and set the vault path to `/Users/name/ZETTL/ZETTLKST ` and use the search function I get the iCloud folder link as results:

![Screenshot 2021-01-20 at 09 30 48](https://user-images.githubusercontent.com/2156115/105148285-8129e200-5b02-11eb-9183-4ccf1813695b.png)

This wil in turn throw an error when I try to open the file:

![Screenshot 2021-01-20 at 09 33 32](https://user-images.githubusercontent.com/2156115/105148427-a6b6eb80-5b02-11eb-853f-d80684837b61.png)

**Comment by macedotavares** [comment]
That's weird. The error message shows that the workflow correctly retrieved the file's URL.
Have you tried setting `vault_path` to `/Users/name/Library/Mobile Documents/com~apple~CloudDocs/ZETTL/ZETTLKST`? Would that even be an acceptable workaround?

**Comment by KingOfSpades** [comment]
Forgot to include that. It would be a acceptable workaround but changing the `vault_path` to `Users/name/Library/Mobile Documents/com~apple~CloudDocs/ZETTL/ZETTLKS` seems to break the search:

![Screenshot 2021-01-20 at 11 10 23](https://user-images.githubusercontent.com/2156115/105160276-58104e00-5b10-11eb-9499-9c9d7021e31f.png)
![Screenshot 2021-01-20 at 11 10 27](https://user-images.githubusercontent.com/2156115/105160293-5e062f00-5b10-11eb-807e-5d963cf96178.png)

**Comment by macedotavares** [comment]
You have to include the slash at the beginning. `/Users/name/Library/Mobile Documents/com~apple~CloudDocs/ZETTL/ZETTLKS`

If that still doesn't work, please activate the debugging mode in the upper right corner of the workflow window, try running the action again and send me a (redacted) copy of the logs.

**Comment by KingOfSpades** [comment]
Ow, that was a typo. The `/` was included. Debugging info is:

```
[10:44:44.899] Obsidian Utilities[Script Filter] Queuing argument 't'
[10:44:45.175] Obsidian Utilities[Script Filter] Script with argv 't' finished
[10:44:45.179] Obsidian Utilities[Script Filter] {"items":[{},{"type":"file","title":"","subtitle":"","arg":""}]}
[10:44:45.183] Obsidian Utilities[Script Filter] Queuing argument 'te'
[10:44:45.429] Obsidian Utilities[Script Filter] Script with argv 'te' finished
[10:44:45.432] Obsidian Utilities[Script Filter] {"items":[{},{"type":"file","title":"","subtitle":"","arg":""}]}
[10:44:45.434] Obsidian Utilities[Script Filter] Queuing argument 'tes'
[10:44:45.506] Obsidian Utilities[Script Filter] Queuing argument 'test'
[10:44:45.687] Obsidian Utilities[Script Filter] Script with argv 'tes' finished
[10:44:45.691] Obsidian Utilities[Script Filter] {"items":[{},{"type":"file","title":"","subtitle":"","arg":""}]}
[10:44:45.937] Obsidian Utilities[Script Filter] Script with argv 'test' finished
[10:44:45.941] Obsidian Utilities[Script Filter] {"items":[{},{"type":"file","title":"","subtitle":"","arg":""}]}
```

I think this is just a limitation of the iCloud file system :/

**Comment by macedotavares** [comment]
OK. Let's check another thing. Maybe those tildes in your vault path are the culprits.

Please edit the script filter and replace the first 3 lines with the following:

```
originalpath="$vault_path"
expanded="${originalpath/#\~/$HOME}"
```

Let me know that solves it.

**Comment by tdfunk** [comment]
FWIW, I worked around this by adding a symbolic link to my iCloud Obsidian vault using iTerm (Terminal would work, too). 

For the OP, that would look like this:

```bash
$ cd ~/Documents
$ ln -s ~/Library/Mobile\ Documents/com\~apple\~CloudDocs/ZETTL/ZETTLKST ./ZETTLKST 
```
Then use `/Users/name/Documents/ZETTLKST` in the vault path setting.

This works for other shell scripts and utilities, as well. 

Bonus: it's also easier to type, making it easier to navigate to vault like this:

```bash
$ cd ~/Documents/ZETTLKST
```

Also, I added an alias for the link to my `zshrc` (or use `bashrc`) file:
```bash
$ alias ov='cd ~/Documents/ZETTLKST'
```

Now getting to my iCloud Obsidian vault takes 3 keystrokes (including `Enter`):

```bash
$ ov
$ pwd 
/Users/name/Documents/ZETTLKST
```

Hope this helps.
